Orange/widgets/tests/test_workflows.py

```diff
@@ -0,0 +1,55 @@
+from os import listdir
+from os.path import isfile, join, dirname, exists
+from itertools import chain
+import logging
+
+from Orange.canvas.application import workflows
+from Orange.canvas.scheme import widgetsscheme
+from Orange.canvas.scheme.readwrite import scheme_load
+from Orange.widgets.tests.base import WidgetTest
+from Orange.canvas.scheme.readwrite import UnknownWidgetDefinition
+
+log = logging.getLogger(__name__)
+
+
+def discover_workflows(tests_dir):
+    assert exists(tests_dir)
+    ows_path = join(tests_dir, "workflows")
+    assert exists(ows_path)
+    ows_files = [f for f in listdir(ows_path)
+                 if isfile(join(ows_path, f)) and f.endswith(".ows")]
+    for ows_file in ows_files:
+        yield join(ows_path, ows_file)
+
+TEST_WORKFLOWS = chain(
+    [t.abspath() for t in workflows.example_workflows()],
+    discover_workflows(dirname(__file__))
+)
+
+
+class TestWorkflows(WidgetTest):
+    @staticmethod
+    def assertCanBeLoaded(filename):
+        new_scheme = widgetsscheme.WidgetsScheme()
+        with open(filename, "rb") as f:
+            try:
+                scheme_load(new_scheme, f)
+            except UnknownWidgetDefinition as e:
+                log.error("Unknown widget '%s' while loading '%s'.",
+                          str(e), filename,
+                          exc_info=True)
+                raise
+            except Exception:
+                log.error("Error occurred while loading '%s'.",
+                          filename,
+                          exc_info=True)
+                raise
+
+    def test_scheme_examples(self):
+        """
+        Test if Orange workflow examples can be opened. Examples in canvas
+        and also those placed "workflows" subfolder.
+        GH-2240
+        """
+        for ows_file in TEST_WORKFLOWS:
+            self.assertCanBeLoaded(ows_file)
```

Comment:
as this test is pretty short now, what about moving the content of the assert method back to the test?